### DISCUSSION TITLE
import existing data and include autogenerated primary key in UI?

### DISCUSSION THREAD

**Discussion by jsjoeio** [supabase/supabase#2543] [Questions] [answered]
Is it possible to add existing content to a table and include an automatically generated key like id?

https://user-images.githubusercontent.com/3806031/126906529-799f3264-21ed-44e8-b7d5-9c266433032b.mp4

_Originally asked [on Twitter](https://twitter.com/jsjoeio/status/1419315208422453249?s=20), cross-posting here_

**Comment by jsjoeio** [comment]
I did get one answer from a supabase fan who suggested doing it manually:

> Seems like you might have to touch good ole reliable SQL to populate the existing records and create the new column as a PK with the default. Would be curious to see if it is possible from the UI though  :)

— @darkfadr

**Reply by silentworks** [reply]
Yeah doing it manually is the way to go about this. You could probably open a issue on github as a feature request to have this in the UI for a future release.

**Reply by jsjoeio** [reply]
Good call! Maybe I will.

**Reply by jsjoeio** [reply]
Done: https://github.com/supabase/supabase/issues/2572

**Comment by jsjoeio** [comment, marked as answer]
Pretty much the same as the other answer, but different order:

> We should add this to the UI but for now you can do it in 2 steps
> 1. paste data (without ID)
> 2. add the column after using the SQL editor

```sql
alter table 
  table_name
add column id 
  serial primary key;
```

> If you want to use uuid then change "serial" to
extensions.uuid_generate_v4()

From @kiwicopple [on Twitter](https://twitter.com/kiwicopple/status/1419452622017232897?s=20)
